media-libs/noise-suppression-for-voice/noise-suppression-for-voice-1.03.ebuild

```
@@ -0,0 +1,47 @@
+# Copyright 2020-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="A real-time noise suppression plugin for voice"
+HOMEPAGE="https://github.com/werman/noise-suppression-for-voice"
+
+if [ "${PV}" = "9999" ]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/werman/noise-suppression-for-voice.git"
+else
+	KEYWORDS="~amd64 ~x86"
+	SRC_URI="https://github.com/werman/noise-suppression-for-voice/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+fi
+
+LICENSE="GPL-3+"
+SLOT="0"
+IUSE="+ladspa lv2 vst vst3"
+
+COMMON_DEPEND="x11-libs/libX11 x11-libs/libXcursor x11-libs/libXinerama x11-libs/libXrandr"
```

I'd split this into:
```suggestion
	COMMON_DEPEND="
		x11-libs/libX11
		x11-libs/libXcursor
		x11-libs/libXinerama
		x11-libs/libXrandr
	"
```

... but also, DEPEND + RDEPEND are almost identical here. And I'm suspicous of freetype being pure-RDEPEND.

So, COMMON_DEPEND="
....
"

then
DEPEND="${COMMON_DEPEND}"
RDEPEND="
	${DEPEND}
	media-libs/freetype
"

If freetype is actually a DEPEND too (I suspect it is, unless it's dlopen-only), then you can just make it RDEPEND="${DEPEND}".

---

`qa-vdb` only listed `media-libs/freetype` as missing in RDEPEND originally. I've also looked at Arch's PKGBUILD and they also only list freetype as a runtime dep, and the X11 libs as make deps.

---

qa-vdb can tell you if something is missing from RDEPEND (it checks ELF NEEDED), but it can't tell you if the header is used or not; i.e. a missing RHS dep in qa-vdb is always a problem, but the LHS needs investigation

---

Okay I've had a quick grep through the source code, and it seems that JUCE indeed loads the X11 libraries via `dlopen`, so they should be in `RDEPEND` as well. freetype also seems to need to be in DEPEND (again looking at the code, it seems they can either use a bundled copy of the header, which I can't find, or the actual freetype headers).